src/shared/sam/debugger/awsSamDebugConfiguration.ts

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+export interface AwsSamDebuggerConfiguration extends vscode.DebugConfiguration {
+    readonly invokeTarget: AwsSamDebuggerInvokeTargetCodeFields | AwsSamDebuggerInvokeTargetTemplateFields
```

Comment:
When we write code that handles the AwsSamDebuggerCongifuration (and sub-structures), we may have an easier time working from a common interface. We won't be able to perform instanceof checks on this property, so if AwsSamDebuggerInvokeTargetCodeFields and AwsSamDebuggerInvokeTargetTemplateFields had a parent interface (which only contained a `target` string field), we'll be able to do some flow control easier.

Reply:
Hmm, yeah, I was thinking we'd still be able to safeguard with just this but the linter isn't a fan: 
![image](https://user-images.githubusercontent.com/29374703/76105574-0ad3ce00-5f8a-11ea-9450-2672b9ef4b60.png)

Will change it.